backend/module/chaos/serverexperimentation/xds/xds.go

```diff
@@ -215,3 +229,47 @@ func (c *rtdsCallbacks) OnFetchRequest(context.Context, *gcpDiscoveryV3.Discover
 func (c *rtdsCallbacks) OnFetchResponse(*gcpDiscoveryV3.DiscoveryRequest, *gcpDiscoveryV3.DiscoveryResponse) {
 	c.logger.Debugw("RTDS OnFetchResponse")
 }
+
+// ECDS Callbacks
+type ecdsCallbacks struct {
+	callbacksBase
+
+	// Track all the seen ECDS resources globally across all the streams. This allows us to query all the requested
+	// resources and present a default value for all the ones that don't have a specific value.
+	// This allows us to set a default value for all the dynamic ECDS resources for all clusters, relying on go-control-plane
+	// to only respond with ones actually requested by the client.
+	requestedECDSResources sync.Map
+}
+
+func (c *ecdsCallbacks) OnStreamOpen(ctx context.Context, streamID int64, typeURL string) error {
+	c.logger.Debugw("ECDS onStreamOpen", "streamID", streamID, "typeURL", typeURL)
+	return c.onStreamOpen(ctx)
+}
+
+func (c *ecdsCallbacks) OnStreamClosed(streamID int64) {
+	c.logger.Debugw("ECDS onStreamClosed", "streamID", streamID)
+	c.onStreamClosed(streamID)
+}
+
+func (c *ecdsCallbacks) OnStreamRequest(streamID int64, req *gcpDiscoveryV3.DiscoveryRequest) error {
+	c.requestedECDSResources.Store(req.Node.Cluster, req.ResourceNames)
```

Comment:
Tracking this per cluster is a bit tricky, I'm not sure if this works as is. For example, if a service is deploying a change that adds a new ECDS resource, two independent Envoys might be sending requests with different sets of resources, causing this per cluster value to flap.

I'm also not sure if Envoy properly combines all ECDS resources into one request (it doesn't for EDS in certain cases), so we should probably verify that (integration test would be nice here). 

The simplest thing to do is just track all the values seen globally, but with some work we can get this working per cluster, lmk which one you prefer.